commit 0ff4a32032059c3bcfe9cae6696ddab4356be972
Author: John Arthorne <johna>
Date:   Fri Dec 5 15:28:39 2008 +0000

    Bug 257431 Query.perform doesn't check for short circuit from collector

2	1	bundles/org.eclipse.equinox.p2.core/src/org/eclipse/equinox/internal/provisional/p2/query/Query.java
diff --git a/bundles/org.eclipse.equinox.p2.core/src/org/eclipse/equinox/internal/provisional/p2/query/Query.java b/bundles/org.eclipse.equinox.p2.core/src/org/eclipse/equinox/internal/provisional/p2/query/Query.java
index 1506649..d633ba9 100644
--- a/bundles/org.eclipse.equinox.p2.core/src/org/eclipse/equinox/internal/provisional/p2/query/Query.java
+++ b/bundles/org.eclipse.equinox.p2.core/src/org/eclipse/equinox/internal/provisional/p2/query/Query.java
@@ -46,7 +46,8 @@ public abstract class Query {
 		while (iterator.hasNext()) {
 			Object candidate = iterator.next();
 			if (isMatch(candidate))
-				result.accept(candidate);
+				if (!result.accept(candidate))
+					break;
 		}
 		return result;
 	}
commit 4fd1b6980ede03f3472ddb5a09b8599216884fdb
Author: John Arthorne <johna>
Date:   Fri Dec 12 21:32:52 2008 +0000

    Bug 257431 Query.perform doesn't check for short circuit from collector

2	1	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/query/QueryableProfileRegistry.java
2	1	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/query/QueryableUpdates.java
2	1	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/QueryableArtifactRepositoryManager.java
2	1	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/QueryableMetadataRepositoryManager.java
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/query/QueryableProfileRegistry.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/query/QueryableProfileRegistry.java
index 91b5ce9..b27fa92 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/query/QueryableProfileRegistry.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/query/QueryableProfileRegistry.java
@@ -36,7 +36,8 @@ public class QueryableProfileRegistry implements IQueryable {
 		try {
 			for (int i = 0; i < profiles.length; i++) {
 				if (query.isMatch(profiles[i]))
-					result.accept(profiles[i]);
+					if (!result.accept(profiles[i]))
+						break;
 				sub.worked(1);
 			}
 		} finally {
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/query/QueryableUpdates.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/query/QueryableUpdates.java
index 939261c..db34a51 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/query/QueryableUpdates.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/query/QueryableUpdates.java
@@ -56,7 +56,8 @@ public class QueryableUpdates implements IQueryable {
 				if (monitor.isCanceled())
 					return result;
 				if (query.isMatch(allUpdates.get(i)))
-					result.accept(allUpdates.get(i));
+					if (!result.accept(allUpdates.get(i)))
+						break;
 				monitor.worked(totalWork / 2 / allUpdates.size());
 			}
 		} catch (OperationCanceledException e) {
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/QueryableArtifactRepositoryManager.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/QueryableArtifactRepositoryManager.java
index f3bb809..6258509 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/QueryableArtifactRepositoryManager.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/QueryableArtifactRepositoryManager.java
@@ -64,7 +64,8 @@ public class QueryableArtifactRepositoryManager implements IQueryable {
 		monitor.beginTask(ProvUIMessages.QueryableArtifactRepositoryManager_RepositoryQueryProgress, repoLocations.length);
 		for (int i = 0; i < repoLocations.length; i++) {
 			if (query == null || query.isMatch(repoLocations[i]))
-				result.accept(repoLocations[i]);
+				if (!result.accept(repoLocations[i]))
+					break;
 			monitor.worked(1);
 		}
 		monitor.done();
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/QueryableMetadataRepositoryManager.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/QueryableMetadataRepositoryManager.java
index 9c01c0a..a427c11 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/QueryableMetadataRepositoryManager.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/QueryableMetadataRepositoryManager.java
@@ -88,7 +88,8 @@ public class QueryableMetadataRepositoryManager implements IQueryable {
 			if (sub.isCanceled())
 				return result;
 			if (query == null) {
-				result.accept(repoLocations.get(i));
+				if (!result.accept(repoLocations.get(i)))
+					break;
 				sub.worked(2);
 			} else {
 				URI location = (URI) repoLocations.get(i);
